library/datasort.c

```diff
@@ -1106,6 +1121,13 @@ static struct datasort_chunk *datasort_merge(struct datasort_ctl *ds_ctl)
 			merged_chunk->fd, merged_chunk->count, merged_chunk->offset, merged_chunk->path);
 
 	datasort_destroy_chunks(dcfg, &ds_ctl->sorted_chunks);
+
+	eblob_stat_set(ds_ctl->cfg->b->stat, EBLOB_GST_DATASORT_SIZE_LIVE_DATA, live_datasize);
```

Comment:
we call this function multiple times for each defrag. Shouldn't it be `eblob_stat_inc` instead of `eblob_stat_set`?

Reply:
This stat is per defrag, so it is zeroed before each defrag.

Reply:
nope, this function is called multiple times per defrag, and most of the times it holds only one bctl (it can hold several ones in case we need to merge them), but we need stat per defrag as it is done for othe counters.

Reply:
Got it, you right.

Reply:
fixed